packages/@react-aria/tag/src/useTag.ts

```diff
@@ -54,30 +54,27 @@ export function useTag(props: AriaTagProps): TagAria {
     }
   }
   const pressProps = {
-    onPress: e => onRemove(children, e)
+    onPress: e => onRemove && onRemove(children, e)
   };
 
   let domProps = filterDOMProps(props);
   return {
-    tagProps: mergeProps(domProps, {
-      'aria-selected': !isDisabled && isSelected,
-      'aria-invalid': validationState === 'invalid' || undefined,
+    tagProps: mergeProps(domProps, rowProps, {
       'aria-errormessage': props['aria-errormessage'],
       onKeyDown: !isDisabled && isRemovable ? onKeyDown : null,
-      role: role === 'gridcell' ? 'row' : null,
-      tabIndex: isDisabled ? -1 : 0
+      tabIndex: isDisabled ? -1 : 0,
+      ref: tagRef
     }),
-    labelProps: {
+    labelProps: mergeProps(gridCellProps, {
       id: tagId,
-      role
-    },
+      ref: labelRef
+    }),
     clearButtonProps: mergeProps(pressProps, {
       'aria-label': removeString,
       'aria-labelledby': `${buttonId} ${tagId}`,
       id: buttonId,
       title: removeString,
-      isDisabled,
-      role
```

Comment:
how'd this role get removed? should it be coming from something like gridCellProps? or is it already inside something with the right role?

Reply:
The clear button is not currently being rendered inside an element with the correct `role`.